tests/TestHarness/transactions.py

```diff
@@ -51,7 +54,7 @@ def createAccount(self, account, creatorAccount, stakedDeposit=1000, waitForTran
             self.trackCmdTransaction(trans)
             transId=NodeosQueries.getTransId(trans)
 
-        return self.waitForTransBlockIfNeeded(trans, waitForTransBlock, exitOnError=exitOnError)
+        return trans
```

Comment:
In the case where the `waitForTransBlockIfNeeded` failed, we used to get log messages and exit if `exitOnError` was set:

```
Utils.cmdError("transaction with id %s never made it into a block" % (transId))
Utils.errorExit("Failed to find transaction with id %s in a block before timeout" % (transId))
```
Seems like these features are gone with the new implementation and may want to consider including them if it is common practice to search for those in logs when something goes bad? And no longer exiting with error code.

Reply:
`exitOnError` was being handled by `processCleosCommand` instead of `waitForTransBlockIfNeeded`.  Unfortunately it doesn't do anything if `silentErrors=True`.  This question prompted rewriting this entire PR, so thanks for that.  (No sarcasm.  The tests are better now.)

Reply:
Modified some tests to disable `silentErrors` to be sure this code path is used at least once.